actionpack/lib/action_dispatch/testing/assertions/response.rb

```diff
@@ -85,17 +86,13 @@ def normalize_argument_to_redirection(fragment)
           end
         end
 
-        def generate_response_message(type)
-          message = "Expected response to be a <#{type}>, but was"
-
+        def response_message
           if @response.redirection?
             redirect_is = normalize_argument_to_redirection(@response.location)
-            message << " a redirect to <#{redirect_is}>"
+            "<302> redirected to <#{redirect_is}>"
```

Comment:
I'd be fine with always showing the response code, and then appending the location if it's a `redirection?` (and it has a location).